test/batchReceiver.spec.ts

```diff
@@ -213,4 +213,157 @@ describe("ReceiveBatch from Queue/Subscription", function(): void {
 
     await receivedMsgs[0].complete();
   });
+
+  it("Abandoning a message 10 times, you get it no more. You can then only get it from the dead letter queue", async function(): Promise<
+    void
+  > {
+    await queueClient.send(testMessages[0]);
+    let abandonMsgCount = 0;
+    let receivedMsgs;
+
+    while (abandonMsgCount < 10) {
+      receivedMsgs = await queueClient.receiveBatch(1);
+
+      should.equal(receivedMsgs.length, 1);
+      should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+      should.equal(receivedMsgs[0].deliveryCount, abandonMsgCount);
+      abandonMsgCount++;
+
+      await receivedMsgs[0].abandon();
+    }
+
+    await testPeekMsgsLength(queueClient, 0);
+
+    const deadLetterQueuePath = process.env.QUEUE_NAME + "/$DeadLetterQueue";
+    const client = namespace.createQueueClient(deadLetterQueuePath);
+    receivedMsgs = await client.receiveBatch(1);
+
+    should.equal(Array.isArray(receivedMsgs), true);
+    should.equal(receivedMsgs.length, 1);
+    should.equal(receivedMsgs[0].body, testMessages[0].body);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+
+    await receivedMsgs[0].complete();
+
+    await testPeekMsgsLength(client, 0);
+    await client.close();
+  });
+
+  it("Abandoning a message 10 times, you get it no more. You can then only get it from the dead letter subscription", async function(): Promise<
+    void
+  > {
+    await topicClient.send(testMessages[0]);
+    let abandonMsgCount = 0;
+    let receivedMsgs;
+    let clientSubscription: SubscriptionClient;
+
+    while (abandonMsgCount < 10) {
+      receivedMsgs = await subscriptionClient.receiveBatch(1);
+
+      should.equal(receivedMsgs.length, 1);
+      should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+      should.equal(receivedMsgs[0].deliveryCount, abandonMsgCount);
+      abandonMsgCount++;
+
+      await receivedMsgs[0].abandon();
+    }
+
+    await testPeekMsgsLength(subscriptionClient, 0);
+
+    const deadLetterTopicPath =
+      process.env.TOPIC_NAME +
+      "/Subscriptions/" +
+      process.env.SUBSCRIPTION_NAME +
+      "/$DeadLetterQueue";
```

Comment:
Let's create a new helper method for this in namespace.ts file, just like we have `getDeadLetterQueuePathForQueue` for queues